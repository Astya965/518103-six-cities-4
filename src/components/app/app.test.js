import React from "react";
import renderer from "react-test-renderer";

import App from "./app.jsx";

const testOffers = [
  {
    id: `1`,
    title: `Test name-1`,
  },
  {
    id: `2`,
    title: `Test name-2`
  }
];

it(`Render App`, () => {
  const tree = renderer.
    create(
        <App offers={testOffers}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();

});
