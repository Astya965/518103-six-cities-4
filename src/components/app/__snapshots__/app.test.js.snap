// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render App with currentOffer 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Offer photo"
            className="property__image"
            src="img/apartment-01.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Offer photo"
            className="property__image"
            src="img/apartment-02.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Test name-1
          </h1>
          <button
            className="property__bookmark-button button property__bookmark-button--active"
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            2
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            3
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            55
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Heating
            </li>
            <li
              className="property__inside-item"
            >
              Kitchen
            </li>
            <li
              className="property__inside-item"
            >
              Frigde
            </li>
            <li
              className="property__inside-item"
            >
              Cable TV
            </li>
            <li
              className="property__inside-item"
            >
              Washing machine
            </li>
            <li
              className="property__inside-item"
            >
              Coffee machine
            </li>
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-max.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Maximum
            </span>
          </div>
        </div>
        <div
          className="property__description"
        >
          <p
            className="property__text"
          >
            A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
          </p>
          <p
            className="property__text"
          >
            The building is green and from 18th century.
          </p>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              3
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/avatar-max.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Leo
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2010-04-27"
                >
                  April 27, 2010
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/avatar-max.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Melon
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2016-08-28"
                >
                  August 28, 2016
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/avatar-angelina.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Su
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "0%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Didn't like that the bathroom door inside the room is wide open.
    Overall the hotel is really close to the metro/train station and it was clean and comfortable.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2017-04-17"
                >
                  April 17, 2017
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            method="post"
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                className="form__rating-input visually-hidden"
                id="5-stars"
                name="rating"
                type="radio"
                value="5"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="4-stars"
                name="rating"
                type="radio"
                value="4"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="3-stars"
                name="rating"
                type="radio"
                value="3"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="2-stars"
                name="rating"
                type="radio"
                value="2"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                className="form__rating-input visually-hidden"
                id="1-star"
                name="rating"
                type="radio"
                value="1"
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-star"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              id="review"
              name="review"
              placeholder="Tell how was your stay, what you like and what can be improved"
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                 and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="places__list
        near-places__list"
      >
        <article
          className="cities__place-card place-card"
          onPointerEnter={[Function]}
          onPointerLeave={[Function]}
        >
          
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="img/apartment-02.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  250
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button "
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "40%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="#"
                onClick={[Function]}
              >
                Test name-2
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;

exports[`Render App Render App without currentOffer 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
              onClick={[Function]}
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
              onClick={[Function]}
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Paris
          </b>
          <form
            action="#"
            className="places__sorting"
            method="get"
          >
            <span
              className="places__sorting-caption"
            >
              Sort by 
            </span>
            <span
              className="places__sorting-type"
              tabIndex="0"
            >
              Popular
              <svg
                className="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlinkHref="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              className="places__options places__options--custom
        places__options--opened"
            >
              <li
                className="places__option places__option--active"
                tabIndex="0"
              >
                Popular
              </li>
              <li
                className="places__option "
                tabIndex="0"
              >
                Price: low to high
              </li>
              <li
                className="places__option "
                tabIndex="0"
              >
                Price: high to low
              </li>
              <li
                className="places__option "
                tabIndex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            className="places__list
        cities__places-list tabs__content"
          >
            <article
              className="cities__place-card place-card"
              onPointerEnter={[Function]}
              onPointerLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-02.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      155
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button place-card__bookmark-button--active"
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="#"
                    onClick={[Function]}
                  >
                    Test name-3
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Hotel
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <div
              id="map"
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
